lib/module.js

```diff
@@ -484,13 +484,30 @@ function tryModuleLoad(module, filename) {
   }
 }
 
-Module._resolveFilename = function(request, parent, isMain) {
+Module._resolveFilename = function(request, parent, isMain, options) {
   if (NativeModule.nonInternalExists(request)) {
     return request;
   }
 
   var paths = Module._resolveLookupPaths(request, parent, true);
```

Comment:
Seems like this should not be executed if `options.prepend` or `options.append` is true